.github/workflows/aws_openshift_rosa_hcp_single_region_tests.yml

```diff
@@ -0,0 +1,643 @@
+                      secret/data/products/infrastructure-experience/ci/common AWS_ACCESS_KEY;
+                      secret/data/products/infrastructure-experience/ci/common AWS_SECRET_KEY;
+                      secret/data/products/infrastructure-experience/ci/common RH_OPENSHIFT_TOKEN;
+                      secret/data/products/infrastructure-experience/ci/common CI_OPENSHIFT_MAIN_PASSWORD;
+                      secret/data/products/infrastructure-experience/ci/common CI_OPENSHIFT_MAIN_USERNAME;
+
+            - name: Add profile credentials to ~/.aws/credentials
+              shell: bash
+              run: |
+                  aws configure set aws_access_key_id ${{ steps.secrets.outputs.AWS_ACCESS_KEY }} --profile ${{ env.AWS_PROFILE }}
+                  aws configure set aws_secret_access_key ${{ steps.secrets.outputs.AWS_SECRET_KEY }} --profile ${{ env.AWS_PROFILE }}
+                  aws configure set region ${{ env.AWS_REGION }} --profile ${{ env.AWS_PROFILE }}
+
+            # Also remove the versioning
+            - name: Create ROSA cluster and login
+              uses: ./.github/actions/aws-openshift-rosa-hcp-single-region-create
+              id: create_cluster
+              # Do not interrupt tests; otherwise, the Terraform state may become inconsistent.
+              if: always() && success()
+              with:
+                  rh-token: ${{ steps.secrets.outputs.RH_OPENSHIFT_TOKEN }}
+                  cluster-name: ${{ matrix.distro.clusterName }}
+                  admin-username: ${{ steps.secrets.outputs.CI_OPENSHIFT_MAIN_USERNAME }}
+                  admin-password: ${{ steps.secrets.outputs.CI_OPENSHIFT_MAIN_PASSWORD }}
+                  aws-region: ${{ env.AWS_REGION }}
+                  s3-backend-bucket: ${{ env.S3_BACKEND_BUCKET }}
+                  s3-bucket-region: ${{ env.S3_BUCKET_REGION }}
+                  openshift-version: ${{ matrix.distro.version }}
+                  tf-modules-revision: ${{ github.head_ref }}
+
+            - name: Export kubeconfig and encrypt it # this is required to pass matrix outputs securely using artifacts
+              id: export_kube_config
+              run: |
+                  # shellcheck disable=SC2005
+                  echo "$(kubectl config view --raw)" > kubeconfig.yaml 2>/dev/null
+                  openssl enc -aes-256-cbc -salt -in kubeconfig.yaml -out encrypted_kubeconfig.enc -pass pass:"${GITHUB_TOKEN}" -pbkdf2
+                  encrypted_kubeconfig_base64=$(base64 -w 0 encrypted_kubeconfig.enc)
+                  echo "kubeconfig_raw=${encrypted_kubeconfig_base64}" >> "$GITHUB_OUTPUT"
+
+            ## Write for matrix outputs workaround
+            - uses: cloudposse/github-action-matrix-outputs-write@ed06cf3a6bf23b8dce36d1cf0d63123885bb8375 # v1
+              id: out
+              with:
+                  matrix-step-name: ${{ github.job }}
+                  matrix-key: ${{ matrix.distro.name }}
+                  outputs: |-
```

Comment:
we are certain that no secret values are persisted that bad actors could use?

Reply:
nice and the token only exists for the duration of its runtime and not possible to extract it.